### DISCUSSION TITLE
SOFA as WebAssembly within JS applications

### DISCUSSION THREAD

**Discussion by OBMartin** [sofa-framework/sofa#4473] [Ideas / Suggestions]
Hello,

I'm currently developing a huge web application where I would like to add CAD Modeling based on SOFA in it. 
Within the Wiki I found an article which informs that Alex is trying to compile SOFA with Emscripten to an WebAssembly, so that it could be Integrated into a rendering browser engine. Source: [https://github.com/sofa-framework/sofa/wiki/Report-of-%22SOFA-dev-meetings%22#06092023](url)

My question would be if there are any results on that topic? Would it be possible to use? or do you see any bigger issues with it? 

Thank you in advance. 

Best regards

Martin

**Comment by alxbilger** [comment]
Hi,

It was actually a student project I supervised and it just ended. We learned some things, but there is no concrete output.

One of the problem that has been encountered is that SOFA has been designed so modules are dynamically linked. Dynamic linking with Emscripten with CMake is not that easy (let me know if you are knowledgeable about this). Therefore, there is currently a work to support static linking in SOFA (for example https://github.com/sofa-framework/sofa/pull/4429).

From my experience, there is no big challenges in compiling SOFA with Emscripten, but it requires some engineering time.
